Packs/Phishing/Scripts/DeleteReportedEmail/DeleteReportedEmail.py

```diff
@@ -0,0 +1,378 @@
+from CommonServerPython import *
+from typing import Callable, Union
+import time
+import traceback
+
+seconds = time.time()
+
+EMAIL_INTEGRATIONS = ['Gmail', 'EWSO365', 'EWS v2', 'Agari Phishing Defense', 'MicrosoftGraphMail',
+                      'SecurityAndCompliance']
+
+
+class MissingEmailException(Exception):
+    def __init__(self):
+        super().__init__('Email was not found in mailbox. It is possible that the email was already deleted manually.')
+
+
+class DeletionFailed(Exception):
+    pass
+
+
+class DeletionArgs:
+
+    @staticmethod
+    def gmail(search_result: dict, search_args: dict):
+        """
+        Parse the arguments needed for the delete operation for Gmail integration.
+        Args:
+            search_result: Results from the priorly preformed search operation
+            search_args: The args used for the search operation
+
+        Returns:
+            The args needed for the deletion operation
+
+        """
+        is_permanent = True if search_args['delete-type'] == 'hard' else False
+        gmail_message_id = search_result[0].get('id')
+        return {
+            'user-id': search_args['user-id'],
+            'message-id': gmail_message_id,
+            'permanent': is_permanent,
+            'using-brand': search_args['using-brand'],
+        }
+
+    @staticmethod
+    def msgraph(search_result: dict, search_args: dict):
+        """
+        Parse the arguments needed for the delete operation for O365 - MSGraph integration.
+        Args:
+            search_result: Results from the priorly preformed search operation
+            search_args: The args used for the search operation
+
+        Returns:
+            The args needed for the deletion operation
+
+        """
+        results = search_result[0].get('value', [])
+        results = [res for res in results if res.get('internetMessageId') == search_args['message-id']]
+        if not results:
+            raise MissingEmailException()
+        internal_id = results[0].get('id')
+        return {
+            'user_id': search_args['user_id'],
+            'message_id': internal_id,
+            'using-brand': search_args['using-brand'],
+        }
+
+    @staticmethod
+    def agari(search_args: dict):
+        """
+        Parse the arguments needed for the delete operation for Agari Phishing Defence integration.
+        Args:
+            search_args: The args used for the search operation
+
+        Returns:
+            The args needed for the deletion operation
+
+        """
+        agari_message_id = demisto.get(demisto.context(), 'incident.apdglobalmessageid')
```

Comment:
Fixed to demisto.incident()
we need to use the incident field since Agari is the only integration that doesn't require a search before the delete operation, hence there is no search results data to work with.